polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5180,7 +5202,11 @@ void MLIRASTConsumer::HandleDeclContext(DeclContext *DC) {
          externLinkage) ||
         emitIfFound.count(name)) {
       // Functions are created in the host context by default.
-      functionsToEmit.emplace_back(*fd, FunctionContext::Host);
+      FunctionToEmit F{*fd, FunctionContext::Host};
+      LLVM_DEBUG(llvm::dbgs()
+                 << __LINE__ << ": Pushing " << F.context << " function "
+                 << fd->getNameAsString() << " to functionsToEmit\n");
+      functionsToEmit.push_back(F);
```

Comment:
No need to get rid of `emplace_back`.

Reply:
emplace_back is not needed because there is nothing to construct.

Reply:
Yes, of course, I'm proposing not creating the instance in line 5205 and creating it in place.

Reply:
I need the instance though because I referenced at line 5282